packages/analyzer/src/javascript/esutil.ts

```diff
@@ -514,18 +514,25 @@ export function getConstructorMethod(
   if (statement) {
     const method = toScannedMethod(
         statement, document.sourceRangeForNode(statement)!, document);
-    if (astNode.id) {
-      method.return = {...method.return, type: astNode.id.name};
+    const typeTag = jsdoc.getTag(jsdoc.parseJsdoc(getAttachedComment(statement) || ''), 'type');
```

Comment:
Why `@type` instead of `@return`? Is that an existing thing? Otherwise I'd think we'd want to just treat it as a method.